v2/spanner-to-sourcedb/src/test/java/com/google/cloud/teleport/v2/templates/SpannerToCassandraLTBase.java

```diff
@@ -0,0 +1,121 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.templates;
+
+import com.google.common.io.Resources;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import org.apache.beam.it.cassandra.CassandraResourceManager;
+import org.apache.beam.it.common.utils.ResourceManagerUtils;
+import org.apache.beam.it.gcp.storage.GcsResourceManager;
+
+/**
+ * Base class for Spanner to cassandra Load tests. It provides helper functions related to
+ * environment setup and assertConditions.
+ */
+public class SpannerToCassandraLTBase extends SpannerToSourceDbLTBase {
+
+  public CassandraResourceManager cassandraResourceManager;
+  public static final String SOURCE_SHARDS_FILE_NAME = "input/cassandra-config.conf";
+
+  public void setupResourceManagers(
+      String spannerDdlResource, String cassandraDdlResource, String artifactBucket)
+      throws IOException {
+    spannerResourceManager = createSpannerDatabase(spannerDdlResource);
+    spannerMetadataResourceManager = createSpannerMetadataDatabase();
+    cassandraResourceManager = generateKeyspaceAndBuildCassandraResource();
+
+    gcsResourceManager =
+        GcsResourceManager.builder(artifactBucket, getClass().getSimpleName(), CREDENTIALS).build();
+    createCassandraSchema(cassandraResourceManager, cassandraDdlResource);
+    createAndUploadCassandraConfigToGcs(gcsResourceManager, cassandraResourceManager);
+    pubsubResourceManager = setUpPubSubResourceManager();
+    subscriptionName =
+        createPubsubResources(
+            getClass().getSimpleName(),
+            pubsubResourceManager,
+            getGcsPath(artifactBucket, "dlq", gcsResourceManager)
+                .replace("gs://" + artifactBucket, ""));
+  }
+
+  protected CassandraResourceManager generateKeyspaceAndBuildCassandraResource() {
+    /* The default is Cassandra 4.1 image. TODO: Explore testing with non 4.1 tags. */
+
+    /* Max Cassandra Keyspace is 48 characters. Base Resource Manager adds 24 characters of date-time at the end.
+     * That's why we need to take a smaller subsequence of the testName.
+     */
+    String uniqueId = testName.substring(0, Math.min(20, testName.length()));
+    return CassandraResourceManager.builder(uniqueId).build();
+  }
+
+  public void cleanupResourceManagers() {
+    ResourceManagerUtils.cleanResources(
+        spannerResourceManager,
```

Comment:
Check once if no errors are thrown if any of this is null. In the future more resource managers  may be used that are part of only some tests that require teardown